Mathlib/Data/Rat/Floor.lean

```diff
@@ -87,6 +87,48 @@ theorem cast_fract (x : ℚ) : (↑(fract x) : α) = fract (x : α) := by
   simp only [fract, cast_sub, cast_intCast, floor_cast]
 #align rat.cast_fract Rat.cast_fract
 
+section NormNum
+
+open Mathlib.Meta.NormNum Qq
+
+theorem isNat_intFloor {R} [LinearOrderedRing R] [FloorRing R] (r : R) (m : ℕ) :
+    IsNat r m → IsNat ⌊r⌋ m := by rintro ⟨⟨⟩⟩; exact ⟨by simp⟩
+
+theorem isInt_intFloor {R} [LinearOrderedRing R] [FloorRing R] (r : R) (m : ℤ) :
+    IsInt r m → IsInt ⌊r⌋ m := by rintro ⟨⟨⟩⟩; exact ⟨by simp⟩
+
+theorem isInt_intFloor_ofIsRat [CharZero α] (r : α) (n : ℤ) (d : ℕ) :
```

Comment:
As mentioned on Zulip, this `CharZero` requirement comes for free once you import `Algebra.Order.Ring.CharZero`. I really dislike for basic inheritance typeclasses to be hidden in obscure files, so I have opened #13654.

Reply:
Can you merge master now that #13654 has landed?

Reply:
Done